src/riak_ql_ddl.erl

```diff
@@ -179,22 +180,44 @@ get_local_key(?DDL{table = T}=DDL, Obj)
     Mod = make_module_name(T),
     get_local_key(DDL, Obj, Mod).
 
+-spec desc_adjusted_key([any()], module(), ?DDL{}) ->
```

Comment:
Not sure we're talking about the same thing. I was suggesting a separation of concerns: call `lk_to_pk` if you need a partition key, call `negate_if_desc` no matter what.